### DISCUSSION TITLE
Civet as a virtual lang via vscode rather than compiled

### DISCUSSION THREAD

**Discussion by PatrickJS** [DanielXMoore/Civet#1080] [Ideas]
Adoption would be way higher if Civet was just a vscode plugin. The idea is that you can view any js/tsx/json file as Civet and when you write/code all it's doing is updating the js/ts/json file with the correct js/ts/json. We do this already for Markdown where we view the previewed version and any edits update the markdown file in-place.

**Comment by PatrickJS** [comment]
I looked into this. it's basically really hard 😂 vscode doesn't provide enough APIs for this. You're forced to create your own editor

**Comment by STRd6** [comment]
It might be possible with [ts2civet](https://github.com/esthedebeste/ts2civet) and a lot of additions to the Civet LSP. We already do tons of compiling and source mapping so it might not be completely unachievable but it is a bit of work for sure.

**Comment by PatrickJS** [comment]
awesome 🎉

**Comment by PatrickJS** [comment]
ok I think this is possible now after some more research

**Comment by adrianhunter** [comment]
https://volarjs.dev/

Might be interesting here. I already successfully used it to create my own civet extension. Also look up the vscode extension from astrojs, it's build with volarjs
